java-rmi/test/com/baeldung/rmi/JavaRMITest.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.rmi;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.rmi.NotBoundException;
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+
+import org.junit.Test;
+
+public class JavaRMITest {
+	
+	@Test
+	public void WhenRunServer_ServerIsStarted() {
+		
+		try {
+			MessengerServiceImpl server = new MessengerServiceImpl();
+			server.createStubAndbind();
+			assertTrue(true);
+		} catch (RemoteException e) {
+			fail("Exception Occured");
+		}
+	}
+	
+	@Test
+	public void WhenClientSendsMessageToServer_ServerSendsResponseMessage() {
+		
+		try {
+			Registry registry = LocateRegistry.getRegistry();	
+			MessengerService server = (MessengerService) registry.lookup("MessengerService");			
+			String responseMessage = server.sendMessage("Client Message");			
+			String expectedMessage = "Server Message";			
+			assertEquals(responseMessage, expectedMessage);
```

Comment:
The java doc says that expected goes first. Also, that will bring it into alignment with your post copy.

Reply:
According to code guidelines. Though there needs to be one line before the expected. Will do that.